raiden/storage/sqlite.py

```diff
@@ -292,6 +307,88 @@ def get_latest_state_change_by_data_field(
 
         return result
 
+    def _get_state_changes(
+            self,
+            limit: int = None,
+            offset: int = None,
+            filters: Dict[str, Any] = None,
+    ) -> List[StateChangeRecord]:
+        """ Return a batch of state change records (identifier and data)
+
+        The batch size can be tweaked with the `limit` and `offset` arguments.
+
+        Additionally the returned state changes can be optionally filtered with
+        the `filters` parameter to search for specific data in the state change data.
+        """
+        limit, offset = _sanitize_limit_and_offset(limit, offset)
+        cursor = self.conn.cursor()
+
+        query = 'SELECT identifier, data FROM state_changes '
+
+        where_clauses = []
+        args = []
+        if filters:
+            for field, value in filters.items():
+                where_clauses.append('json_extract(data, ?)=?')
+                args.append(f'$.{field}')
+                args.append(value)
+
+            query += (
+                f"WHERE {' AND '.join(where_clauses)}"
+            )
+
+        query += 'ORDER BY identifier ASC LIMIT ? OFFSET ?'
+        args.append(limit)
+        args.append(offset)
+
+        cursor.execute(query, args)
+        result = []
+        try:
+            rows = cursor.fetchall()
+            for row in rows:
+                result.append(StateChangeRecord(
+                    state_change_identifier=row[0],
+                    data=row[1],
+                ))
+        except AttributeError:
+            raise InvalidDBData(
+                'Your local database is corrupt. Bailing ...',
+            )
+
+        return result
+
+    def batch_query_state_changes(
+            self,
+            batch_size: int,
```

Comment:
I'd default "batch_size" parameter as in the it doesn't matter too much how big are chuncks as long as they are reasonable.

Reply:
Thank you but I prefer explicitness, especially if it's just used in one place.